### PR TITLE
refactor: Port MaHidden to typescript

### PR DESCRIPTION
**Pull request by edufarre**

### CODE REVIEW

File: src/components/MaHidden/MaHidden.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    return slots().default &&
+      shouldRenderContent(
+        props.above,
+        props.below,
+        parent.$layout.currentBreakpoint
+      )
+      ? h(component, slots().default)
+      : h()
+  },
```
Comment (alex-castelo) [inline src/components/MaHidden/MaHidden.ts:65]: Checkmate Typescript ♟️ 🎩

File: src/components/MaHidden/MaHidden.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      `neither 'below' nor 'above' prop were provided. Please provide at least one`
+    )
+  }
+}
+
+function shouldRenderContent(
+  above: AboveBreakpoints,
+  below: BelowBreakpoints,
+  currentBreakpoint: Breakpoint
+) {
```
Comment (alex-castelo) [inline src/components/MaHidden/MaHidden.ts:81]: I don't know if we have talked about this but I believe that for Vue 2 and TS they recommend explicitly noting the return types of both methods and computed properties.

It is also true that surely, it already makes the correct type inference, so I think we would have to come to an agreement. 🤝

**Review by alex-castelo — APPROVED** [review]
🚀 🚀

File: src/components/MaHidden/MaHidden.ts (hunk 2 above)
Comment (edufarre) [reply src/components/MaHidden/MaHidden.ts:81]: This is not a method inside the component nor computed property, functional components does not have computed props

Comment (alex-castelo) [reply src/components/MaHidden/MaHidden.ts:81]: Right! Sorry, my bad 😅

**Review by sergigraciaj — APPROVED** [review]

**Pull request merged**